### ISSUE TITLE
Add broken package dependencies to a custom dependencies section in package.json?

### ISSUE THREAD

**Issue by JamesCoyle** [pnpm/pnpm#1962]
In the FAQ, there is a section recommending what to do when a package doesn't correctly declare it's dependencies. 

> **Solution 1**
> In the following example, a dependency does not have the iterall module in its own list of deps.
> 
> The easiest solution to resolve missing dependencies of the buggy packages is to add iterall as a dependency to our project's package.json.
> 
> You can do so, by installing it via:
> ```
> pnpm i iterall
> ```
> It will be automatically added to your project's package.json.
>  ```
>   "dependencies": {
>     ...
>     "iterall": "^1.2.2",
>     ...
>   }
> ```

I feel it would be helpful to have an option to place such dependencies into their own section within `package.json` as they aren't direct dependencies to your project. This then allows you to open an issue or pull request for the broken package and remove the dependency once it is resolved as it is clear which dependencies are needed for your project and which ones were just to fix issues with other packages.

Example:
```console
pnpm i iterall -t
```

```json
"tempDependencies": {
    ...
    "iterall": "^1.2.2",
    ...
}
```

**Comment by mcmxcdev** [comment]
I don't think this needs a custom section since the common practice is already for package authors to use `peerDependencies` to declare which packages are needed. You will be able to auto-install peer dependencies once this PR lands: https://github.com/pnpm/pnpm/pull/4776

**Comment by zkochan** [comment]
This is implemented in current pnpm, so closing.

`packageExtensions` in pnpm-workspace.yaml adds the missing dependency to the package that forgot to declare it, so it no longer has to go into your own dependencies.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
